test/Spec/MockContract.hs

```diff
@@ -258,6 +259,18 @@ runPABEffectPure initState req =
     go (UploadDir dir) = mockUploadDir dir
     go (QueryChainIndex query) = mockQueryChainIndex query
 
+    incSlot :: forall (v :: Type). MockContract w v -> MockContract w v
+    incSlot mc =
+      mc <* modify @(MockContractState w) (tip %~ incTip)
+
+    incTip TipAtGenesis = Tip 1 (BlockId "00") 0
+    incTip Tip {tipSlot, tipBlockId, tipBlockNo} =
+      Tip
+        { tipSlot = tipSlot + 1
+        , tipBlockId = tipBlockId
+        , tipBlockNo = tipBlockNo
+        }
```

Comment:
We can use record dot here!
```suggestion
    incTip t@Tip {} = t{tipSlot + 1}
```

Reply:
I'm afraid this doesn't work with the plugin:
> Using the preprocessor, but not the GHC plugin:
>
>    expr{lbl1.lbl2 = val} is equivalent to expr{lbl1 = (expr.lbl1){lbl2 = val}}, performing a nested update.
>    expr{lbl * val} is equivalent to expr{lbl = expr.lbl * val}, where * can be any operator.
>    expr{lbl1.lbl2} is equivalent to expr{lbl1.lbl2 = lbl2}.
https://github.com/ndmitchell/record-dot-preprocessor

Reply:
Also, the Tip is a sum type, so I couldn't even use a conventional record update syntax either (I have a non-exhaustive pattern match error)

Reply:
Oh I see, thats a shame, disregard then, my bad!